docs/best-practices.md

```diff
@@ -0,0 +1,65 @@
+particular object types, resource types, their paths etc. The document intends
+to provide design guidance. Its rules are not mandatory to be followed.
+
+
+## Object Type Definitions
+
+Define object types to be abstracted from hardware layout and implementation
+details whenever possible. <br />
+*Rationale:* The CSMIM data model structure should be the same for all current
+and future system instantiations, regardless of how a particular manufacturer
+chooses to structure its hardware.
+
+Define object types to capture one piece of functionality. <br />
+*Rationale:* This promotes abstraction from hardware layout. And it avoids the
+situation where the resources of one CSMIM object have to be provided by
+multiple hardware devices - which is complicated to achieve.
+
+
+## Resource Type Definitions
+
+Restrict resource identifiers to use only lower-case characters and the
+underscore. <br />
+*Rationale:* This avoids lower-case/upper-case confusion.
+
+Choose the identifier of a `bool` resource type (or parameter or `dict` item)
+such that the meaning of the *true* and *false* value is immediately obvious.
+
+Use the `enum` data type to capture a *finite* number of states that will be
+processed by another device on the aircraft. Do not declare the enumeration to
+be future-extensible if a processing device may come from a different
+manufacturer. <br />
+*Rationale:* This ensures future compatibility and clarity of software 
+implementations.
+
+Use the `string` data type to capture a potentially infinite number of states.
+Values are typically not processed by another device on the aircraft (displaying
+a string to a human does not constitute "processing").
+
+Use a writable resource if the resource has a single "value" that can be
+changed. Make sure that the operation is idempotent (multiple WRITE requests
+have the same effect as a single one). If this is not the case, you must use an
+executable resource instead.
+
+
+## Paths and Data Model Structure
+
+Choose CSMIM object paths such that the resulting object tree resembles the
+structure of the aircraft and its subsystems, in particular the aircraft cabin
+with its monuments and assemblies. The real-world equivalent of an object lower
+in the tree should always be *contained in* its parent. <br />
+*Rationale:* This ensures future extensibility of the object tree. <br />
+*Example:* Galley inserts are part of a galley. A master galley control unit is
+part of a galley. The object representing a galley insert cannot have its path
+below an object representing the MGCU, because a galley insert is not contained
+in an MGCU.
+
+Restrict object paths to use only lower-case characters and the underscore,
+except for equipment IDs (e.g. Airbus FIN). <br />
+*Rationale:* This avoids lower-case/upper-case confusion.
+
+Use `.../collection/<id>` as an object path element to capture a collection of
+similar objects. The collection name should be a plural noun. <br />
```

Comment:
do you mean `.../<collection>/<id>`? And is the "collection name" from the next sentence the same as the id here?